tests/test_utils/test_linalg/test_inner_product.py

```diff
@@ -0,0 +1,32 @@
+"""Tests for general inner products."""
+
+import numpy as np
+import pytest
+
+from probnum.utils.linalg import euclidean_inprod, euclidean_norm
+
+
+@pytest.fixture(scope="module", params=[1, 10, 50])
+def n(request) -> int:
+    """Vector size."""
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def vector0(n: int) -> np.ndarray:
+    rng = np.random.default_rng(86 + n)
+    return rng.standard_normal(size=(n,))
+
+
+@pytest.fixture(scope="module")
+def vector1(n: int) -> np.ndarray:
+    rng = np.random.default_rng(567 + n)
+    return rng.standard_normal(size=(n,))
+
+
+def test_euclidean_inprod(vector0: np.ndarray, vector1: np.ndarray):
+    assert vector0 @ vector1 == pytest.approx(euclidean_inprod(v=vector0, w=vector1))
+
```

Comment:
It may be useful to test that `euclidean_inprod` correctly computes the inner products of multiple vectors.

I think we could add a test like this:
`@pytest.fixture(scope = "module", params = [1, 3, 5])
def m(request) -> int:
    """ Number of simultaneous vectors """
    return request.param

@pytest.fixture(scope = "module")
def matrix0(n: int, m: int) -> np.ndarray:
    rng = np.random.default_rng()
    return rng.standard_normal(size=(n,m))

@pytest.fixture(scope = "module")
def matrix1(n: int, m: int) -> np.ndarray:
    rng = np.random.default_rng()
    return rng.standard_normal(size=(n,m))

def test_euclidean_inprod_multiple(matrix0: np.ndarray, matrix1: np.ndarray):
    m = matrix0.shape[1]
    inner_list = np.zeros(m)
    for i in range(m):
        inner_list[i] = np.inner(matrix0[i],matrix1[i])
    assert inner_list == pytest.approx(euclidean_inprod(v=matrix0,w=matrix1))
`